From 138cb336108faab12961813bdf7ccc1ae4e1822b Mon Sep 17 00:00:00 2001
From: Leonard Foerster <foersleo@amazon.com>
Date: Wed, 26 Jul 2023 08:40:14 +0000
Subject: [PATCH] linux-firmware: ethernet: Remove firmware for ethernet/IB
 devices

Bottlerocket does not ship drivers for older network equipment. Without
the drivers shipping the firmware for thesde devices does not make
sense. Drop the firmware to reduce image size.

The following list maps driver names as specified in WHENCE to kernel
config options enabling the driver. This way we can easily decide if we
need to add firmware back into the package when we enable new drivers.

* cassini - CONFIG_CASSINI
* slicoss - CONFIG_SLICOSS
* sxg - CONFIG_SXG
* cxgb3 - CONFIG_CHELSIO_T3
* e100 - CONFIG_E100
* acenic - CONFIG_ACENIC

Signed-off-by: Leonard Foerster <foersleo@amazon.com>
---
diff -ruN eth-step1/LICENCE.e100 eth-step2/LICENCE.e100
--- eth-step1/LICENCE.e100	2025-11-18 03:42:28.360774094 +0000
+++ eth-step2/LICENCE.e100	1970-01-01 00:00:00.000000000 +0000
@@ -1,28 +0,0 @@
-Copyright (c) 1999-2001, Intel Corporation
-
-All rights reserved.
-
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions are met:
-
- 1. Redistributions of source code must retain the above copyright notice,
-    this list of conditions and the following disclaimer.
-
- 2. Redistributions in binary form must reproduce the above copyright notice,
-    this list of conditions and the following disclaimer in the documentation
-    and/or other materials provided with the distribution.
-
- 3. Neither the name of Intel Corporation nor the names of its contributors
-    may be used to endorse or promote products derived from this software
-    without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
-AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-DISCLAIMED. IN NO EVENT SHALL CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
-PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
-EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
diff -ruN eth-step1/LICENCE.microchip eth-step2/LICENCE.microchip
--- eth-step1/LICENCE.microchip	2025-11-18 03:42:28.361774096 +0000
+++ eth-step2/LICENCE.microchip	1970-01-01 00:00:00.000000000 +0000
@@ -1,40 +0,0 @@
-Copyright (C) 2018 Microchip Technology Incorporated and its subsidiaries.
-All rights reserved.
-
-REDISTRIBUTION: Permission is hereby granted by Microchip Technology
-Incorporated (Microchip), free of any license fees, to any person obtaining a
-copy of this firmware (the "Software"), to install, reproduce, copy and
-distribute copies, in binary form, hexadecimal or equivalent formats only, the
-Software and to permit persons to whom the Software is provided to do the same,
-subject to the following conditions:
-
-* Any redistribution of the Software must reproduce the above copyright notice,
-  this license notice, and the following disclaimers and notices in the
-  documentation and/or other materials provided with the Software.
-
-* Neither the name of Microchip, its products nor the names of its suppliers
-  may be used to endorse or promote products derived from this Software without
-  specific prior written permission.
-
-* No reverse engineering, decompilation, or disassembly of this Software is
-  permitted.
-
-Limited patent license.  Microchip grants a world-wide, royalty-free,
-non-exclusive, revocable license under any patents that it now has or hereafter
-may have, own or control related to the Software to make, have made, use,
-import, offer to sell and sell ("Utilize") this Software, but solely to the
-extent that any such patent is necessary to Utilize the Software in conjunction
-with Microchip processors. The patent license shall not apply to any other
-combinations which include this Software nor to any other Microchip patents or
-patent rights. No hardware per se is licensed hereunder.
-
-DISCLAIMER: THIS SOFTWARE IS PROVIDED BY MICROCHIP "AS IS" AND ANY EXPRESS OR
-IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
-MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT ARE
-DISCLAIMED. IN NO EVENT SHALL ATMEL BE LIABLE FOR ANY DIRECT, INDIRECT,
-INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
-PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
-LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
-OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
-ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
diff -ruN eth-step1/LICENSE.hfi1_firmware eth-step2/LICENSE.hfi1_firmware
--- eth-step1/LICENSE.hfi1_firmware	2025-11-18 03:42:28.366774108 +0000
+++ eth-step2/LICENSE.hfi1_firmware	1970-01-01 00:00:00.000000000 +0000
@@ -1,39 +0,0 @@
-Copyright (c) 2015, Intel Corporation. 
-All rights reserved. 
-
-Redistribution.
-
-Redistribution and use in binary form, without modification, are permitted
-provided that the following conditions are met: 
-*    Redistributions must reproduce the above copyright notice and the
-     following disclaimer in the documentation and/or other materials provided
-     with the distribution. 
-*    Neither the name of Intel Corporation nor the names of its suppliers may
-     be used to endorse or promote products derived from this software without
-     specific prior written permission. 
-*    No reverse engineering, decompilation, or disassembly of this software is
-     permitted. 
-
-Limited patent license.
-
-Intel Corporation grants a world-wide, royalty-free, non-exclusive license
-under patents it now or hereafter owns or controls to make, have made, use,
-import, offer to sell and sell (“Utilize”) this software, but solely to the
-extent that any such patent is necessary to Utilize the software alone. The
-patent license shall not apply to any combinations which include this software.
-No hardware per se is licensed hereunder. 
-
-
-DISCLAIMER.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
-ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
-WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
-ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
-(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
-ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
-
diff -ruN eth-step1/WHENCE eth-step2/WHENCE
--- eth-step1/WHENCE	2025-11-18 03:42:28.566784439 +0000
+++ eth-step2/WHENCE	2025-11-18 03:42:57.833615812 +0000
@@ -156,67 +156,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: cassini - Sun Cassini
-
-File: sun/cassini.bin
-
-Licence: Unknown
-
-Found in hex form in kernel source.
-
---------------------------------------------------------------------------
-
-Driver: slicoss - Alacritech IS-NIC products
-
-File: slicoss/gbdownload.sys
-File: slicoss/gbrcvucode.sys
-File: slicoss/oasisdbgdownload.sys
-File: slicoss/oasisdownload.sys
-File: slicoss/oasisrcvucode.sys
-
-Licence: Redistributable. See LICENCE.alacritech for details
-
-Found in hex form in kernel source.
-
---------------------------------------------------------------------------
-
-Driver: sxg - Alacritech IS-NIC products
-
-File: sxg/saharadownloadB.sys
-File: sxg/saharadbgdownloadB.sys
-
-Licence: Redistributable. See LICENCE.alacritech for details
-
-Found in hex form in kernel source.
-
---------------------------------------------------------------------------
-
-Driver: cxgb3 - Chelsio Terminator 3 1G/10G Ethernet adapter
-
-File: cxgb3/t3b_psram-1.1.0.bin
-File: cxgb3/t3c_psram-1.1.0.bin
-File: cxgb3/t3fw-7.0.0.bin
-File: cxgb3/t3fw-7.1.0.bin
-File: cxgb3/t3fw-7.4.0.bin
-File: cxgb3/t3fw-7.10.0.bin
-File: cxgb3/t3fw-7.12.0.bin
-
-Licence: GPLv2 or OpenIB.org BSD license, no source visible
-
---------------------------------------------------------------------------
-
-Driver: cxgb3 - Chelsio Terminator 3 1G/10G Ethernet adapter
-
-File: cxgb3/ael2005_opt_edc.bin
-File: cxgb3/ael2005_twx_edc.bin
-File: cxgb3/ael2020_twx_edc.bin
-
-Licence: Redistributable. See LICENCE.cxgb3 for details
-
-Found in hex form in kernel source.
-
---------------------------------------------------------------------------
-
 Driver: cxgb4 - Chelsio Terminator 4/5/6 1/10/25/40/100G Ethernet adapter
 
 File: cxgb4/t4fw-1.14.4.0.bin
@@ -244,28 +183,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: e100 - Intel PRO/100 Ethernet NIC
-
-File: e100/d101m_ucode.bin
-File: e100/d101s_ucode.bin
-File: e100/d102e_ucode.bin
-
-Licence: Redistributable. See LICENCE.e100 for details
-
---------------------------------------------------------------------------
-
-Driver: acenic - Alteon AceNIC Gigabit Ethernet card
-
-File: acenic/tg1.bin
-File: acenic/tg2.bin
-
-Licence: Unknown
-
-Found in hex form in kernel source, but source allegedly available at
-http://alteon.shareable.org/
-
---------------------------------------------------------------------------
-
 Driver: tg3 - Broadcom Tigon3 based gigabit Ethernet cards
 
 File: tigon/tg3.bin
@@ -4055,21 +3972,6 @@
 
 --------------------------------------------------------------------------
 
-Driver: hfi1 - Intel OPA Gen 1 adapter
-
-File: hfi1_dc8051.fw
-Version: 1.27.0
-File: hfi1_fabric.fw
-Version: 0x1055
-File: hfi1_pcie.fw
-Version: 0x4755
-File: hfi1_sbus.fw
-Version: 0x10130001
-
-Licence: Redistributable. See LICENSE.hfi1_firmware for details
-
---------------------------------------------------------------------------
-
 Driver: knav_qmss_queue - TI Keystone 2 QMSS driver
 
 File: ti-keystone/ks2_qmss_pdsp_acc48_k2_le_1_0_0_9.bin
